### DISCUSSION TITLE
Support for unmaintained NodeJS versions?

### DISCUSSION THREAD

**Discussion by karfau** [xmldom/xmldom#495] [Polls]
Multiple tools we are using for development stopped supporting the node versions we are currently supporting, and which we are checking our code against in GitHub Actions.

Currently we support Node v10 and above.
But development has been happening with node 12 for quite a while.
As more and more major updates of devDependencies start requiring Node version 14 or 16, I'm considering to "drop support" for anything below Node v14 and start using 14 or even 16 for development.

Dropping support would mean that we no longer run our code against those versions on GitHub Actions, and wouldn't notice when we would break things.
So it doesn't mean we will rewrite our codebase in ES modules (any time soon), but of course making node 14 the minimum supported version would make such a change possible, see #214 .

With the upcoming 0.9.0 we have the option to do such a change, but I don't want to do it without hearing back from users.

https://github.com/nodejs/release#release-schedule

Please also leave a comment why/for how long you need to use that version of node, if the one you picked is no longer maintained.

**Comment by Ponynjaa** [comment]
I think there's no reason to support Node versions that are already EOL. Whoever uses these Node versions should be forced to update due to other issues already.
